### PR TITLE
Use standard dirent types instead of custom ones

### PR DESCRIPTION
**Pull request by agkaminski**
JIRA: RTOS-603

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: ext2/dir.c [hunk 1]
```diff
@@ -153,16 +153,20 @@ int _ext2_dir_read(ext2_t *fs, ext2_obj_t *dir, offs_t offs, struct dirent *res,
...
+
 	case DIRENT_BLKDEV:
-		res->d_type = dtDev;
+		res->d_type = DT_BLK;
 		break;
 
 	default:
-		res->d_type = dtFile;
+		res->d_type = DT_REG;
+		break;
```
[the remark this answers is not in the document]
Comment (agkaminski) [reply ext2/dir.c]: -.-

File: dummyfs/dir.c [hunk 2]
```diff
@@ -145,16 +145,24 @@ int dir_add(dummyfs_t *ctx, dummyfs_object_t *dir, const char *name, uint32_t mo
...
-	else if (S_ISREG(mode))
-		n->type = otFile;
-	else if (S_ISCHR(mode) || S_ISBLK(mode))
-		n->type = otDev;
-	else if (S_ISLNK(mode))
-		n->type = otSymlink;
-	else
-		n->type = otUnknown;
+	if (S_ISDIR(mode)) {
+		n->type = DT_DIR;
```
Comment (lukileczo) [inline dummyfs/dir.c:149]: `#include <dirent.h>` :smiley:

Comment (agkaminski) [reply dummyfs/dir.c:150]: oops, thx

File: dummyfs/dir.c [hunk 3]
```diff
@@ -17,6 +17,7 @@
 #include <sys/file.h>
```
Comment (lukileczo) [inline dummyfs/dir.c:17]: I think this include can be removed now?

**Review by lukileczo — APPROVED** [review]

**Review by badochov — APPROVED** [review]
LGTM

**Pull request merged**